plugins/net_plugin/net_plugin.cpp

```diff
@@ -315,6 +315,86 @@ namespace eosio {
    constexpr uint32_t signed_block_which       = fc::get_index<net_message, signed_block>();       // see protocol net_message
    constexpr uint32_t packed_transaction_which = fc::get_index<net_message, packed_transaction>(); // see protocol net_message
 
+   class connections_manager {
+      alignas(hardware_destructive_interference_size)
+      mutable std::shared_mutex connections_mtx;
+      std::set<connection_ptr>  connections;
+      chain::flat_set<string>   supplied_peers;
+
+      alignas(hardware_destructive_interference_size)
+      std::mutex                            connector_check_timer_mtx;
+      unique_ptr<boost::asio::steady_timer> connector_check_timer;
+
+      /// thread safe, only modified on startup
+      std::chrono::milliseconds                                heartbeat_timeout{def_keepalive_interval*2};
+      fc::microseconds                                         max_cleanup_time;
+      boost::asio::steady_timer::duration                      connector_period{0};
+      uint32_t                                                 max_client_count{def_max_clients};
+      std::function<void(net_plugin::p2p_connections_metrics)> update_p2p_connection_metrics;
+
+   private: // must call with held mutex
+      connection_ptr find_connection_impl(const string& host) const;
+      void add_impl(connection_ptr&& c);
+      void connect_impl(const string& peer);
+
+      void connection_monitor(const std::weak_ptr<connection>& from_connection);
+
+   public:
+      void add_supplied_peers(const vector<string>& peers ) {
+         std::lock_guard g(connections_mtx);
+         supplied_peers.insert( peers.begin(), peers.end() );
+      }
+
+      // not thread safe, only call on startup
+      void init( std::chrono::milliseconds heartbeat_timeout_ms,
```

Comment:
Maybe remove the spaces after parenthesis for consistency (and alignment with next parameters)
```suggestion
      void init(std::chrono::milliseconds heartbeat_timeout_ms,
```

Reply:
Done